### PR TITLE
Migrate uncertainties from ESSsans

### LINKED ISSUE #16: Move variance broadcast handling utils from ESSsans
- Note scipp/esssans#101.
- Note that the event-data version may make assumptions that are not correct in the non-SANS workflows.

### PR DESCRIPTION
**Pull request by nvaytet**
Fixes #16

### CODE REVIEW

File: src/ess/reduce/uncertainty.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        if dim not in data.dims:
+            size *= dim_size
+    data = data.copy()
+    data.variances *= size
+    return data.broadcast(sizes={**sizes, **data.sizes}).copy()
+
+
+def drop_variances_if_broadcast(
+    data: Union[sc.Variable, sc.DataArray], sizes: Dict[str, int]
+) -> Union[sc.Variable, sc.DataArray]:
```
Comment (SimonHeybrock) [inline src/ess/reduce/uncertainty.py:53]: No `@overload`s here, unlike above?

**Review by SimonHeybrock — COMMENTED** [review]
Is the plan do add the event-mode functions later?

**Comment by nvaytet** [comment]
> Is the plan do add the event-mode functions later?

Yes. I opened #20 to keep track of it.

File: tests/uncertainty_test.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    assert_identical(broadcast_with_upper_bound_variances(var, {'z': 1}), expected)
+    expected = sc.ones(dims=['z'], shape=[2]) * values
+    expected.variances = 2 * expected.values
+    assert_identical(broadcast_with_upper_bound_variances(var, {'z': 2}), expected)
+    expected = sc.ones(dims=['z'], shape=[2]) * values
+    expected.variances = 2 * expected.values
+    assert_identical(
+        broadcast_with_upper_bound_variances(var, {'y': 3, 'z': 2}),
+        expected.transpose(['y', 'z', 'x']),
+    )
```
Comment (SimonHeybrock) [inline tests/uncertainty_test.py:38]: Missing tests for drop and fail?

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**